### PR TITLE
Feature/qap 1191 run pre commit

### PR DESCRIPTION
**Pull request by diasdm**
Run pre-commit for Python and ROS workflows.

For ROS tested here:
* Without config: https://github.com/MOV-AI/movai_DELETE_ME_DAVID/runs/8015010476?check_suite_focus=true
* With config resulting in failure: https://github.com/MOV-AI/movai_DELETE_ME_DAVID/runs/8015093309?check_suite_focus=true
* With config resulting in success: https://github.com/MOV-AI/movai_DELETE_ME_DAVID/runs/8015182584?check_suite_focus=true

For Python tested here:
* Without config (expected success): https://github.com/MOV-AI/metadata-static-analysis/runs/8019970456?check_suite_focus=true
* With config resulting in failure: https://github.com/MOV-AI/metadata-static-analysis/runs/8020033380?check_suite_focus=true
* With config resulting in success: https://github.com/MOV-AI/metadata-static-analysis/runs/8020108960?check_suite_focus=true

### CODE REVIEW

**Comment by diasdm** [comment]
These changes should only impact the few people who have already started using pre-commit for Python or ROS, @MithunKinarullathil and I.

In the following repos:
* https://github.com/MOV-AI/metadata-static-analysis - no impact 
* https://github.com/MOV-AI/movai_ot_ottobo - no impact
* https://github.com/MOV-AI/movai_common - no impact
* https://github.com/MOV-AI/movai-developer-tools - impact, already asked Mithun to correct

File: .github/workflows/py-workflow.yml [hunk 1]
```diff
@@ -54,6 +54,16 @@ jobs:
...
 
+    - name: Check if pre-commit config exists
+      id: check_files
+      uses: andstor/file-existence-action@v1
+      with:
+        files: ".pre-commit-config.yaml"
+
+    - name: Run pre-commit
+      if: steps.check_files.outputs.files_exists == 'true'
+      uses: pre-commit/action@v3.0.0
```
Comment (AlexFernandes-MOVAI) [inline .github/workflows/py-workflow.yml:65]: notes from the maintainer:

> DEPRECATED this action is in maintenance-only mode and will not be accepting new features.
> 
> Please switch to using [pre-commit.ci](https://pre-commit.ci/) which is faster and has more features.

**Review by AlexFernandes-MOVAI — APPROVED** [review]
action used for pre-commit is in maintenance mode, but it seems OK for our need

File: .github/workflows/py-workflow.yml (hunk 1 above)
Comment (diasdm) [reply .github/workflows/py-workflow.yml:65]: Yeah, the action is so simple that if needed we can even copy it and lose the dependency. Also pre-commit.ci is being run a SAAS, which, at least for me doesn't make much sense.

**Pull request merged**